pkg/operator/ceph/pool/controller.go

```diff
@@ -419,64 +420,91 @@ func deletePool(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo,
 	return nil
 }
 
-// remove removes any element from a list
-func remove(list []string, s string) []string {
-	for i, v := range list {
-		if v == s {
-			list = append(list[:i], list[i+1:]...)
+// generateStatsPoolList combines existingStatsPools and rookStatsPools, removes items in removePools,
+// removes duplicates, ensures no empty strings, and returns a comma-separated string in a deterministic order.
+func generateStatsPoolList(existingStatsPools []string, rookStatsPools []string, removePools []string) string {
+	// Create a map to keep track of pools that should be removed
+	removeMap := make(map[string]struct{})
+	for _, pool := range removePools {
+		if pool != "" {
+			removeMap[pool] = struct{}{}
 		}
 	}
 
-	return list
+	// Create a list to store unique pools
```

Comment:
This comment is probably unnecessary for understanding the function

Reply:
deleted